File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0)
            return false;

        String num = "" + x;

        int l = 0 , r = num.length() - 1;

        while(l < r){
            if(num.charAt(l) != num.charAt(r))
                return false;

            l++;
            r--;
        }

        return true;

    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        s = str(x)

        lft , rght = 0 , len(s) - 1

        while(lft < rght) :
            if s[lft] != s[rght] :
                return False

            lft += 1
            rght -= 1

        return True

